pkgs/development/python-modules/notify-py/default.nix

```diff
@@ -1,33 +1,38 @@
-{ lib, buildPythonPackage, fetchPypi, isPy3k, alsaUtils, libnotify, which, loguru, pytest }:
+{ lib, stdenv, buildPythonPackage, fetchPypi, isPy3k, alsaUtils, libnotify
+, which, loguru, pytestCheckHook }:
 
 buildPythonPackage rec {
-  pname = "notify_py";
+  pname = "notify-py";
   version = "0.3.1";
 
   disabled = !isPy3k;
 
   src = fetchPypi {
-    inherit pname version;
+    inherit version;
+    pname = "notify_py";
     sha256 = "5ba696d18ffe1d7070f3d0a5b4923fee4d6c863de6843af105bec0ce9915ebad";
   };
 
-  postPatch = ''
-   substituteInPlace setup.py \
-     --replace "loguru==0.4.1" "loguru~=0.5.0"
-  '';
-
-  propagatedBuildInputs = [ alsaUtils libnotify loguru which ];
+  propagatedBuildInputs = [ loguru which ]
+    ++ lib.optionals stdenv.isLinux [ alsaUtils libnotify ];
 
-  checkInputs = [ alsaUtils libnotify pytest which ];
+  checkInputs = [ pytestCheckHook which ]
+    ++ lib.optionals stdenv.isLinux [ alsaUtils libnotify ];
 
-  checkPhase = ''
-    pytest
+  # Tests search for "afplay" binary which is built in to MacOS and not available in nixpkgs
+  preCheck = ''
+    mkdir $TMP/bin
+    cp /usr/bin/afplay $TMP/bin
+    export PATH="$TMP/bin:$PATH"
```

Comment:
I don't know whether this is clever, awful, or both but this makes the tests pass on macOS, I even hear the ding sound play.

Reply:
Yeah, this is probably a bad idea. Having a $TMP/bin is fine, but this will break when sandboxed.

Can you depend on ${coreutils}/bin/true and symlink that to $TMP/bin/afplay instead?

Reply:
Yeah I knew this would break the sandbox. The workaround you're suggesting is just to get the tests to pass, right? Make `afplay` a noop basically?

Reply:
Pretty much, yes. Having a binary that runs but doesn't actually do anything.

Reply:
Changed to that method.